### ISSUE TITLE
yowsup-cli: quick repeated presence subscribes cause exception

### ISSUE THREAD

**Issue by yniwly43** [tgalal/yowsup#1554]
Tried using yowsup-cli to figure out why the blocking happens.
So, I connected, did a /presence subscribe for around 30 contacts, quickly. (Pasted /presence subscribe <number> 30 times into console). I got an exception, below:

```
yowsup-cli  v2.0.15
yowsup      v2016.4.26.2

Copyright (c) 2012-2016 Tarek Galal
http://www.openwhatsapp.org

This software is provided free of charge. Copying and redistribution is
encouraged.

If you appreciate this software and you would like to support future
development please consider donating:
http://openwhatsapp.org/yowsup/donate


tx: <iq xmlns="w:p" type="get" id="1"></iq>

Auth: Logged in!

Yowsup Cli client
==================
Type /help for available commands

[connected]:
:tx: <presence to="963442044164@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963405610710@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963446887998@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963409039597@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963443010186@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963444828874@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963428722089@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963405819488@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963406007403@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963424222916@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963448882100@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963423423864@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963444514614@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963444775652@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963445772747@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963407484014@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963426662445@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963405064130@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963408870055@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963424335758@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963447787801@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963445743214@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963426998224@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963447456588@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963424260654@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963428888556@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963444918079@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963428888531@s.whatsapp.net" type="subscribe"></presence>

[connected]
:tx: <presence to="963403048779@s.whatsapp.net" type="subscribe"></presence>

[connected]
:

rx: <stream:error><xml-not-well-formed></xml-not-well-formed></stream:error>

Traceback (most recent call last):
  File "/usr/lib/python2.7/logging/__init__.py", line 851, in emit
    msg = self.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 724, in format
    return fmt.format(record)
  File "/usr/lib/python2.7/logging/__init__.py", line 464, in format
    record.message = record.getMessage()
  File "/usr/lib/python2.7/logging/__init__.py", line 324, in getMessage
    msg = str(self.msg)
  File "/home/yniwly43/yowsup/yowsup/layers/auth/protocolentities/stream_error.py", line 46, in __str__
    out += self.getErrorData()
TypeError: cannot concatenate 'str' and 'dict' objects

```

Seems like a threading issue to me, since it only sometimes happens.
Yowsup is not threadsafe, so.. it makes sense when you add a thread. :) 

Edit: it seems to me that the error is actually two different errors:
1. Several messages got "mixed" due to the copy/paste, so there was bad XMPP.. and then we got the xml-not-well-formed message.
2. getErrorData() in StreamErrorProtocolEntity should always return a string.
3. Generally, it would be better to use string formatting instead of string concatenation.. less error prone.

**Comment by jlguardi** [comment]
Related with #1518

**Comment by tgalal** [comment]
Fixed in ac2dd59bb65eb6e18343400506beafdc34a78e44
